ssg/build_cpe.py

```diff
@@ -264,10 +266,15 @@ def enrich_with_cpe_info(self, cpe_products):
 
     def pass_parameters(self, product_cpes):
         if self.arg:
-            associated_cpe_item_as_dict = product_cpes.get_cpe(self.cpe_name).represent_as_dict()
+            cpe = product_cpes.get_cpe(self.cpe_name)
+            parameters = cpe.args[self.arg]
+            parameters.update(self.as_dict())
+            associated_cpe_item_as_dict = cpe.represent_as_dict()
             new_associated_cpe_item_as_dict = apply_formatting_on_dict_values(
-                associated_cpe_item_as_dict, self.as_dict())
+                associated_cpe_item_as_dict, parameters)
             new_associated_cpe_item_as_dict["id_"] = self.as_id()
+            if cpe.is_templated():
+                new_associated_cpe_item_as_dict["template"]["vars"] = parameters
```

Comment:
I have checked the `build` directory and I found that the compiled CPE, eg. `build/rhel9/cpe_items/package_ntp.yml` looks ugly:

````
name: cpe:/a:ntp
check_id: cond_package_ntp
bash_conditional: ''
ansible_conditional: ''
is_product_cpe: false
args:
    ntp:
        pkgname: ntp
        title: NTP daemon and utilities
        id: package_ntp
        name: package
        arg: ntp
title: Package ntp is installed
definition_location: /home/jcerny/work/git/scap-security-guide/shared/applicability/package.yml
template:
    name: cond_package
    vars:
        pkgname: ntp
        title: NTP daemon and utilities
        id: package_ntp
        name: package
        arg: ntp
documentation_complete: true
````

This is a resolved CPE Item so the `args` key probably doesn't make sense there.

Reply:
This code build templates.

Reply:
The problem here is that there is no distinction between resolved and unresolved CPEe, they all are stored in the same `product_cpes.cpes_as_id` dictionary.

Anyhow, this is the best place to fill *template* variables, IMHO.

Reply:
This is again a design smell and this byproduct is, unlike other byproducts, confusing. First of all, some arguments repeated twice (or is it a coincidence?), and second of all, they are already partially substituted, as Jan has noticed. Finally, how come that e.g. `bash_conditional` is there and it is empty? Is the template incomplete?

Reply:
> , how come that e.g. bash_conditional is there and it is empty? Is the template incomplete?

Because the snippet that I pasted is from build/rhel9/cpe_items/package_ntp.yml which is an interim file that is created before the templates are resolved. The platforms are stored in `build/<product>/platforms`  instead. But there the bash conditional is empty as well because the template doesn't contain any bash check. And that leads me to the question to @evgenyz:  Why the template submitted in this PR doesn't contain any Bash check? Can we add it? Will that need some additional Python code to be implemented?

Reply:
It is empty because `shared/applicability/package.yml` does not have `bash_conditional` at this moment. Yes, this is another set of changes that is pending.

Reply:
aha

Reply:
What is the reason that `bash_conditional` is empty then? Is the template incomplete, or do they get populated at some point later? If it is the case of the incomplete template, then please add those bits, as reviewing an incomplete template doesn't feel reasonable, as changes to this part will affect the integration of the incoming part as well.

Reply:
Added missing `*_conditional`s to the `package` CPE. 

The `args` element is now empty in the resolved entity.